controllers/infra/drivers/aliyun/create_instance.go

```diff
@@ -203,7 +203,7 @@ func getDataDisks(host *v1.Hosts) []ecs.RunInstancesDataDisk {
 }
 
 func getHostName(hosts *v1.Hosts, infra *v1.Infra) string {
-	return fmt.Sprintf("%s-%s-%d", infra.Namespace, infra.Name, hosts.Index)
+	return fmt.Sprintf("%s-%s-${instance_id}", infra.Namespace, infra.Name)
```

Comment:
What is `${instance_id}` means here?

Reply:
> What is `${instance_id}` means here?

according to aliyun api docs:
<img width="510" alt="image" src="https://user-images.githubusercontent.com/88967335/223019728-f8276da8-597f-419e-858e-7d4492ba350a.png">

Reply:
Use some info that we do not control is not so pretty? 
Imaging this situation: user want to know roles/disks info when only have instance name, the auto assigned ${instance_id} is not saved under infra CR, so they cannot get real instance full name?

Reply:
> Use some info that we do not control is not so pretty? Imaging this situation: user want to know roles/disks info when only have instance name, the auto assigned ${instance_id} is not saved under infra CR, so they cannot get real instance full name?

User can get this instanceId from their infra cr metadata:
<img width="330" alt="image" src="https://user-images.githubusercontent.com/88967335/223041342-ff0f6e2a-9b38-4f48-97ca-d63eeab165fe.png">